_scripts/diff.py

```diff
@@ -0,0 +1,90 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Compare localization files
+"""
+
+from __future__ import annotations
+
+from argparse import ArgumentParser
+from os import walk
+from os.path import commonpath, exists, join, relpath
+from sys import exit
+
+from moz.l10n.resource import UnsupportedResource, l10n_equal, parse_resource
+
+
+def diff_tree(a_root: str, b_root: str, ignore: set[str]) -> int:
+    common_root = commonpath((a_root, b_root))
+    diff_count = 0
+    b_seen: set[str] = set()
+    for dirpath, dirnames, filenames in walk(a_root):
+        dirnames[:] = [dn for dn in dirnames if not dn.startswith((".", "_"))]
+        for fn in filenames:
+            if not fn.startswith((".", "_")):
```

Comment:
Do we even have files starting with `_` that we want to ignore?

Reply:
We have directories starting with `_` to ignore, and it makes more sense to keep the rule for files as well.

Reply:
If that's the case, would it make sense to put the check into a function? e.g. something along the line of

```python
def ignore_global(name: str) -> bool:
	return name.startswith((".", "_"))

dirnames[:] = [dn for dn in dirnames if not ignore_global(dn)]

if not ignore_global(fn):
```

Reply:
Eh, sure, but this is a hacky utility script where code quality and maintainability really aren't paramount. If this thing lives for a long time, I'll refactor it to use `moz.l10n.paths.L10nDiscoverPaths` instead.